### ISSUE TITLE
Release 4.4.0 - External integrations modules

### ISSUE THREAD

**Issue by nico-stefani** [wazuh/wazuh#14374]
The following issue aims to perform all the required testing for the current release candidate to ensure the modules (AWS, GCP, Azure and DockerListener) works as expected, report the results, and open new issues for any encountered errors.

## Modules tests information

|||
|----------------------------------|------ |
| **Main release candidate issue** |   -   |
| **Version**                      | 4.4 |
| **Release candidate #**          | -   |
| **Tag**                          | -  |
| **Previous modules tests issue** | -    |

## Test report procedure

All test results must have one the following statuses: 
|                                  |                                                        |
|-----------------|-------------------------------------------------------------------------|
| :green_circle:  | The test passed successfully. 										    |
| :yellow_circle: | The test passed but some enhancements need to be applied to the module. |
| :red_circle:    | The test failed.                                                        |
| :white_circle:  | The test does not apply.                                                |

Any failing test must be properly addressed with a new issue, detailing the error and the possible cause. It must be included in the `Fixes` section of the current release candidate main issue.

The resulting logs for the tests must be included in the status report so it can be used by the auditors to dig deeper into any possible failures and details.

## Conclusions

| **Module**      | **Status**  |**Notes**|
|-----------------|-------------|---------|
| AWS             | :green_circle: Done | #14402  |
| GCP             | :green_circle: Done | #14440  |
| Azure           | :green_circle: Done| #14439 |
| Docker Listener | :green_circle: Done | #14415  |

## Auditors validation
The definition of done for this one is the validation of the conclusions and the test results from all auditors.

All checks from below must be accepted in order to close this issue.
- [ ] @davidjiglesias 
- [x] @CarlosRS9 
- [x] @nico-stefani

**Comment by nico-stefani** [comment]
## AWS Tests results

A test is considered valid if the following conditions are met:

- The expected result described in the test itself is met.
- The expected amount of events were sent to `analysisd`.
- The expected output was present in the `{wazuh_path}/logs/ossec.log` file.

The data available in our Wazuh-dev environment for each of these supported services will be used. Additional logs may be created if they are necessary or there is insufficient data for testing.


## Test using ossec.conf in a manager

<details><summary>Configuration</summary>

    <wodle name="aws-s3">
    <disabled>no</disabled>
    <remove_from_bucket>no</remove_from_bucket>
    <interval>10m</interval>
    <run_on_start>yes</run_on_start>
    <skip_on_error>no</skip_on_error>

    <bucket type="cloudtrail">
        <aws_profile>dev</aws_profile>
        <name>wazuh-aws-wodle-cloudtrail</name>
        <only_logs_after>2021-NOV-10</only_logs_after>
    </bucket>

    <bucket type="custom">
        <aws_profile>dev</aws_profile>
        <name>wazuh-aws-wodle-macie</name>
        <only_logs_after>2021-NOV-10</only_logs_after>
    </bucket>

    <service type="cloudwatchlogs">
        <aws_profile>dev</aws_profile>
        <aws_log_groups>4_4_test</aws_log_groups>
        <only_logs_after>2021-AUG-20</only_logs_after>
        <regions>us-west-1</regions>
    </service>
  </wodle>
</details>


<details><summary>Output</summary>

    2022/07/21 16:49:29 wazuh-modulesd:aws-s3[17491] wm_aws.c:82 at wm_aws_main(): INFO: Starting fetching of logs.
    2022/07/21 16:49:29 wazuh-modulesd:aws-s3[17491] wm_aws.c:134 at wm_aws_main(): INFO: Executing Bucket Analysis: (Bucket: wazuh-aws-wodle-cloudtrail, Type: cloudtrail, Profile: dev)
    2022/07/21 16:49:29 wazuh-modulesd:aws-s3[17491] wm_aws.c:336 at wm_aws_run_s3(): DEBUG: Create argument list
    2022/07/21 16:49:29 wazuh-modulesd:aws-s3[17491] wm_aws.c:451 at wm_aws_run_s3(): DEBUG: Launching S3 Command: wodles/aws/aws-s3 --bucket wazuh-aws-wodle-cloudtrail --aws_profile dev --only_logs_after 2021-NOV-10 --type cloudtrail --debug 2
    2022/07/21 16:49:32 wazuh-modulesd:aws-s3[17491] wm_aws.c:491 at wm_aws_run_s3(): DEBUG: Bucket:  -  OUTPUT: DEBUG: +++ Debug mode on - Level: 2
    2022/07/21 16:49:32 wazuh-modulesd:aws-s3[17491] wm_aws.c:134 at wm_aws_main(): INFO: Executing Bucket Analysis: (Bucket: wazuh-aws-wodle-macie, Type: custom, Profile: dev)
    2022/07/21 16:49:32 wazuh-modulesd:aws-s3[17491] wm_aws.c:336 at wm_aws_run_s3(): DEBUG: Create argument list
    2022/07/21 16:49:32 wazuh-modulesd:aws-s3[17491] wm_aws.c:451 at wm_aws_run_s3(): DEBUG: Launching S3 Command: wodles/aws/aws-s3 --bucket wazuh-aws-wodle-macie --aws_profile dev --only_logs_after 2021-NOV-10 --type custom --debug 2
    2022/07/21 16:49:35 wazuh-modulesd:aws-s3[17491] wm_aws.c:491 at wm_aws_run_s3(): DEBUG: Bucket:  -  OUTPUT: DEBUG: +++ Debug mode on - Level: 2
    2022/07/21 16:49:37 wazuh-modulesd:aws-s3[17491] wm_aws.c:169 at wm_aws_main(): INFO: Executing Service Analysis: (Service: cloudwatchlogs, Profile: dev)
    2022/07/21 16:49:37 wazuh-modulesd:aws-s3[17491] wm_aws.c:515 at wm_aws_run_service(): DEBUG: Create argument list
    2022/07/21 16:49:37 wazuh-modulesd:aws-s3[17491] wm_aws.c:619 at wm_aws_run_service(): DEBUG: Launching S3 Command: wodles/aws/aws-s3 --service cloudwatchlogs --aws_profile dev --only_logs_after 2021-AUG-20 --regions us-west-1 --aws_log_groups 4_4_test --debug 2
    2022/07/21 16:49:41 wazuh-modulesd:aws-s3[17491] wm_aws.c:660 at wm_aws_run_service(): DEBUG: Service: cloudwatchlogs  -  OUTPUT: DEBUG: +++ Debug mode on - Level: 2
    2022/07/21 16:49:41 wazuh-modulesd:aws-s3[17491] wm_aws.c:174 at wm_aws_main(): INFO: Fetching logs finished.
    2022/07/21 16:49:41 wazuh-modulesd:aws-s3[17491] wm_aws.c:78 at wm_aws_main(): DEBUG: Sleeping until: 2022/07/21 16:59:29

</details>


#### Conclusions

The module worked as expected. Events were sent to the analysis module.



## Test using ossec.conf in an agent

Deploy an Ubuntu 18 agent and install the required dependencies described in the official Wazuh documentation, then run the module adding the following configuration to the ossec.conf using python 3.9:



<details><summary>Configuration</summary>

    <wodle name="aws-s3">
    <disabled>no</disabled>
    <remove_from_bucket>no</remove_from_bucket>
    <interval>10m</interval>
    <run_on_start>yes</run_on_start>
    <skip_on_error>no</skip_on_error>

    <bucket type="cloudtrail">
        <aws_profile>dev</aws_profile>
        <name>wazuh-aws-wodle-cloudtrail</name>
        <only_logs_after>2021-NOV-10</only_logs_after>
    </bucket>

    <bucket type="custom">
        <aws_profile>dev</aws_profile>
        <name>wazuh-aws-wodle-macie</name>
        <only_logs_after>2021-NOV-10</only_logs_after>
    </bucket>

    <service type="cloudwatchlogs">
        <aws_profile>dev</aws_profile>
        <aws_log_groups>4_4_test</aws_log_groups>
        <only_logs_after>2021-AUG-20</only_logs_after>
        <regions>us-west-1</regions>
    </service>
  </wodle>
    
</details>

<details><summary>Output</summary>

    2022/07/21 17:09:55 wazuh-modulesd:aws-s3[2835] wm_aws.c:60 at wm_aws_main(): INFO: Module AWS started
    2022/07/21 17:09:55 wazuh-modulesd:aws-s3[2835] wm_aws.c:82 at wm_aws_main(): INFO: Starting fetching of logs.
    2022/07/21 17:09:55 wazuh-modulesd:aws-s3[2835] wm_aws.c:134 at wm_aws_main(): INFO: Executing Bucket Analysis: (Bucket: wazuh-aws-wodle-cloudtrail, Type: cloudtrail, Profile: dev)
    2022/07/21 17:09:55 wazuh-modulesd:aws-s3[2835] wm_aws.c:336 at wm_aws_run_s3(): DEBUG: Create argument list
    2022/07/21 17:09:55 wazuh-modulesd:aws-s3[2835] wm_aws.c:451 at wm_aws_run_s3(): DEBUG: Launching S3 Command: wodles/aws/aws-s3 --bucket wazuh-aws-wodle-cloudtrail --aws_profile dev --only_logs_after 2021-NOV-10 --type cloudtrail --debug 2
    2022/07/21 17:10:02 wazuh-modulesd:aws-s3[2835] wm_aws.c:491 at wm_aws_run_s3(): DEBUG: Bucket:  -  OUTPUT: DEBUG: +++ Debug mode on - Level: 2
    2022/07/21 17:10:03 wazuh-modulesd:aws-s3[2835] wm_aws.c:134 at wm_aws_main(): INFO: Executing Bucket Analysis: (Bucket: wazuh-aws-wodle-macie, Type: custom, Profile: dev)
    2022/07/21 17:10:03 wazuh-modulesd:aws-s3[2835] wm_aws.c:336 at wm_aws_run_s3(): DEBUG: Create argument list
    2022/07/21 17:10:03 wazuh-modulesd:aws-s3[2835] wm_aws.c:451 at wm_aws_run_s3(): DEBUG: Launching S3 Command: wodles/aws/aws-s3 --bucket wazuh-aws-wodle-macie --aws_profile dev --only_logs_after 2021-NOV-10 --type custom --debug 2
    2022/07/21 17:10:07 wazuh-modulesd:aws-s3[2835] wm_aws.c:491 at wm_aws_run_s3(): DEBUG: Bucket:  -  OUTPUT: DEBUG: +++ Debug mode on - Level: 2
    2022/07/21 17:10:09 wazuh-modulesd:aws-s3[2835] wm_aws.c:169 at wm_aws_main(): INFO: Executing Service Analysis: (Service: cloudwatchlogs, Profile: dev)
    2022/07/21 17:10:09 wazuh-modulesd:aws-s3[2835] wm_aws.c:515 at wm_aws_run_service(): DEBUG: Create argument list
    2022/07/21 17:10:09 wazuh-modulesd:aws-s3[2835] wm_aws.c:619 at wm_aws_run_service(): DEBUG: Launching S3 Command: wodles/aws/aws-s3 --service cloudwatchlogs --aws_profile dev --only_logs_after 2021-AUG-20 --regions us-west-1 --aws_log_groups 4_4_test --debug 2
    2022/07/21 17:10:13 wazuh-modulesd:aws-s3[2835] wm_aws.c:660 at wm_aws_run_service(): DEBUG: Service: cloudwatchlogs  -  OUTPUT: DEBUG: +++ Debug mode on - Level: 2
    2022/07/21 17:10:13 wazuh-modulesd:aws-s3[2835] wm_aws.c:174 at wm_aws_main(): INFO: Fetching logs finished.
    2022/07/21 17:10:13 wazuh-modulesd:aws-s3[2835] wm_aws.c:78 at wm_aws_main(): DEBUG: Sleeping until: 2022/07/21 17:19:55
</details>


#### Conclusions

The module worked as expected. Events were sent to the analysis module.

**Comment by nico-stefani** [comment]
Hey Team,

I run the light test (using `ossec.conf`) for AWS wodle. I was able to verify that the events are sending to analisysd, checking the `WAZUH_PATH/logs/archives/2022/Jul/*` logs.

Some sample of this:
```
{"timestamp":"2022-07-22T11:52:05.261+0000","agent":{"id":"000","name":"4cf109f8f551"},"manager":{"name":"4cf109f8f551"},"id":"1658490725.0","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"DEBUG: +++ Marker: AWSLogs/166157441623/CloudTrail/us-west-1/2022/03/30/166157441623_CloudTrail_us-west-1_20220330T0000Z_HASDoKlxgfdkdIOa.json.txt","decoder":{},"location":"aws-s3"}
{"timestamp":"2022-07-22T11:52:05.302+0000","agent":{"id":"000","name":"4cf109f8f551"},"manager":{"name":"4cf109f8f551"},"id":"1658490725.0","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"DEBUG: +++ Marker: AWSLogs/166157449999/CloudTrail/us-west-1/2021/12/23/166157441623_CloudTrail_us-west-1_20211223T0000Z_HASDOtJxxxxxxxxx.json.gz","decoder":{},"location":"aws-s3"}
{"timestamp":"2022-07-22T12:02:05.234+0000","agent":{"id":"000","name":"4cf109f8f551"},"manager":{"name":"4cf109f8f551"},"id":"1658491325.0","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"DEBUG: +++ Marker: AWSLogs/166157441623/CloudTrail/us-west-1/2022/03/30/166157441623_CloudTrail_us-west-1_20220330T0000Z_HASDoKlxgfdkdIOa.json.txt","decoder":{},"location":"aws-s3"}
{"timestamp":"2022-07-22T12:02:05.274+0000","agent":{"id":"000","name":"4cf109f8f551"},"manager":{"name":"4cf109f8f551"},"id":"1658491325.0","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"DEBUG: +++ Marker: AWSLogs/166157449999/CloudTrail/us-west-1/2021/12/23/166157441623_CloudTrail_us-west-1_20211223T0000Z_HASDOtJxxxxxxxxx.json.gz","decoder":{},"location":"aws-s3"}
{"timestamp":"2022-07-22T12:12:04.681+0000","agent":{"id":"000","name":"4cf109f8f551"},"manager":{"name":"4cf109f8f551"},"id":"1658491924.0","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"DEBUG: +++ Marker: AWSLogs/166157441623/CloudTrail/us-west-1/2022/03/30/166157441623_CloudTrail_us-west-1_20220330T0000Z_HASDoKlxgfdkdIOa.json.txt","decoder":{},"location":"aws-s3"}
{"timestamp":"2022-07-22T12:12:04.722+0000","agent":{"id":"000","name":"4cf109f8f551"},"manager":{"name":"4cf109f8f551"},"id":"1658491924.0","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"DEBUG: +++ Marker: AWSLogs/166157449999/CloudTrail/us-west-1/2021/12/23/166157441623_CloudTrail_us-west-1_20211223T0000Z_HASDOtJxxxxxxxxx.json.gz","decoder":{},"location":"aws-s3"}


{"timestamp":"2022-07-22T11:52:08.243+0000","agent":{"id":"000","name":"4cf109f8f551"},"manager":{"name":"4cf109f8f551"},"id":"1658490728.0","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"DEBUG: +++ Marker: 2022/02/14/firehose_macie-1-2022-02-14-01-20-42-eb1f80bc-5d9d-426b-a0a1-e0ad8d3ef656","decoder":{},"location":"aws-s3"}
{"timestamp":"2022-07-22T12:02:08.329+0000","agent":{"id":"000","name":"4cf109f8f551"},"manager":{"name":"4cf109f8f551"},"id":"1658491328.0","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"DEBUG: +++ Marker: 2022/02/14/firehose_macie-1-2022-02-14-01-20-42-eb1f80bc-5d9d-426b-a0a1-e0ad8d3ef656","decoder":{},"location":"aws-s3"}
{"timestamp":"2022-07-22T12:12:07.590+0000","agent":{"id":"000","name":"4cf109f8f551"},"manager":{"name":"4cf109f8f551"},"id":"1658491927.0","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"DEBUG: +++ Marker: 2022/02/14/firehose_macie-1-2022-02-14-01-20-42-eb1f80bc-5d9d-426b-a0a1-e0ad8d3ef656","decoder":{},"location":"aws-s3"}

```

As stopper, still not have the access to GCP and Azure to continue with these tests. The accounts are requested in https://github.com/wazuh/wazuh-automation/issues/873 and https://github.com/wazuh/wazuh-automation/issues/872

**Comment by nico-stefani** [comment]
# DockerListener tests results

Manual testing was performed to ensure everything works after the change. The tests were carried out using both a Wazuh manager and a Wazuh agent both using the following configuration:
```
<wodle name="docker-listener">
    <interval>30s</interval>
    <attempts>5</attempts>
    <run_on_start>yes</run_on_start>
    <disabled>no</disabled>
</wodle>
```

The [logall_json option](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#logall-json) must be enabled in the `ossec.conf` file before starting the tests.

### Test docker listener using ossec.conf in a manager
Restart Wazuh and check ossec.log to verify the module is running


<details><summary>ossec.log output</summary>

    2022/07/22 09:39:00 wazuh-modulesd:docker-listener[28889] wm_docker.c:45 at wm_docker_main(): INFO: Module docker-listener started.
    2022/07/22 09:39:00 wazuh-modulesd:docker-listener[28889] wm_docker.c:58 at wm_docker_main(): INFO: Starting to listening Docker events.
    2022/07/22 09:39:00 wazuh-modulesd:docker-listener[28889] wm_docker.c:62 at wm_docker_main(): DEBUG: Launching command 'wodles/docker/DockerListener'

</details>



#### Conclusions

The module worked as expected. No errors were present in the ossec.log.
